core/commands/pin.go

```diff
@@ -153,12 +153,10 @@ var listPinCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
 		Tagline: "List objects pinned to local storage",
 		ShortDescription: `
-Returns a list of hashes of objects being pinned. Objects that are indirectly
-or recursively pinned are not included in the list.
+Recursively returns a list of hashes of objects being pinned.
```

Comment:
hmm... this isnt right (and it was misleading before).

Should say something like: 

```
Returns a list of objects that are pinned locally. By default, only recursively pinned objects are 
returned, but others may be show via the '--type' flag.

Example:
    $ echo "hello" | ipfs add -q
    QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN
    $ ipfs pin ls
    QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN
    # now remove the pin, and repin it directly
    $ ipfs pin rm QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN
    $ ipfs pin add -r=false QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN
    $ ipfs pin ls --type=direct
    QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN
```

Maybe just the first sentence in the `ShortDescription` and all of it in the `LongDescription`